packages/web/src/components/userprofile/UserProfile.tsx

```diff
@@ -40,31 +48,43 @@ export const UserProfile: React.FC<UserProfileProps> = ({ user, setUser }: UserP
     setUser(newUser);
   };
 
-  return (
-    <Box border="1px" borderColor="gray.200" boxShadow="base" p={3}>
-      {user.discordId ? (
-        <>
-          <Text fontSize="xl"> Your discord account is linked with: {user.discordId} </Text>
-          <Button size="md" colorScheme="blue" onClick={unlinkDiscordId}>
-            Unlink Discord Account
-          </Button>
-          <Text>{errorMessage}</Text>
-        </>
-      ) : (
-        <>
-          <Text fontSize="xl">
-            {' '}
-            After linking you&apos;ll be able to join our community and talk with others from
-            Project X on a variety of topics.{' '}
-          </Text>
-          <Button as="a" size="sm" colorScheme="blue" href="/api/auth/discord/login">
-            Login with Discord
+  if (editToggle)
+    return (
+      <VStack border="1px" borderColor="gray.200" boxShadow="base" p={3}>
+        <EditUserForm setEditToggle={setEditToggle} setUser={setUser} user={user} />
+      </VStack>
+    );
+  else {
+    return (
+      <VStack border="1px" borderColor="gray.200" boxShadow="base" p={3}>
+        {user.discordId ? (
+          <>
+            <Text fontSize="xl"> Your discord account is linked with: {user.discordId} </Text>
+            <Button size="md" colorScheme="blue" onClick={unlinkDiscordId}>
+              Unlink Discord Account
+            </Button>
+            <Text>{errorMessage}</Text>
+          </>
+        ) : (
+          <>
+            <Text fontSize="xl">
+              After linking you&apos;ll be able to join our community and talk with others from
+              Project X on a variety of topics.{' '}
```

Comment:
Again, be careful with spaces 😉 
```suggestion
              Project X on a variety of topics.
```